plugins/verifier/sbom/sbom.go

```diff
@@ -70,35 +72,54 @@ func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, refe
 
 	ctx := context.Background()
 	referenceManifest, err := referrerStore.GetReferenceManifest(ctx, subjectReference, referenceDescriptor)
-
 	if err != nil {
-		return nil, err
+		return &verifier.VerifierResult{
+			Name:      input.Name,
+			IsSuccess: false,
+			Message:   fmt.Sprintf("Error fetching reference manifest for subject: %s reference descriptor: %v", subjectReference, referenceDescriptor.Descriptor),
+		}, err
 	}
 
+	var mediaType string
 	for _, blobDesc := range referenceManifest.Blobs {
+		mediaType = blobDesc.MediaType
 		refBlob, err := referrerStore.GetBlobContent(ctx, subjectReference, blobDesc.Digest)
-		if err != nil {
-			return nil, err
-		}
 
-		var sbomBlob SbomContents
-		if err := json.Unmarshal(refBlob, &sbomBlob); err != nil {
-			return nil, fmt.Errorf("failed to parse sbom: %v", err)
-		}
-
-		if sbomBlob.Contents == "bad" {
+		if err != nil {
 			return &verifier.VerifierResult{
 				Name:      input.Name,
 				IsSuccess: false,
-				Message:   fmt.Sprintf("SBOM verification failed. The contents is %s.", sbomBlob.Contents),
-			}, nil
+				Message:   fmt.Sprintf("Error fetching blob for subject: %s digest: %s", subjectReference, blobDesc.Digest),
+			}, err
+		}
+
+		switch mediaType {
+		case SpdxJsonMediaType:
+			return ProcessSpdxJsonMediaType(input.Name, refBlob)
+		default:
 		}
 	}
 
 	return &verifier.VerifierResult{
 		Name:      input.Name,
-		IsSuccess: true,
-		Message:   "SBOM verification success. The contents is good.",
+		IsSuccess: false,
+		Message:   fmt.Sprintf("Unsupported mediaType: %s", mediaType),
 	}, nil
+}
 
+func ProcessSpdxJsonMediaType(name string, refBlob []byte) (*verifier.VerifierResult, error) {
+	if ok, err := parser2v2.Load2_2(refBlob); ok != nil {
```

Comment:
nit: maybe we can rename `ok` to `doc`? `ok` sounds like a bool value to me. And I wonder if we should check `err` instead of `doc`.